modules/blocks/index.js

```diff
@@ -64,3 +71,56 @@ export function getBlockSettings( slug ) {
 export function getBlocks() {
 	return Object.values( blocks );
 }
+
+/**
+ * Iterates through children of a given element to discover block nodes.
+ *
+ * @param  {Element}  rootNode Element to iterate
+ * @return {Object[]}          Block nodes
+ */
+export function findBlockNodes( rootNode ) {
+	const nodes = [];
+
+	let child = rootNode.firstChild;
+	while ( child ) {
+		// Test as opening comment
+		if ( Node.COMMENT_NODE !== child.nodeType || ! RE_BLOCK_OPEN.test( child.nodeValue ) ) {
+			child = child.nextSibling;
+			continue;
+		}
+
+		// Found opening block comment
+		const match = child.nodeValue.match( RE_BLOCK_OPEN );
+		const closeTag = '/' + child.nodeValue.trim();
+		const block = {
+			type: match[ 1 ],
+			startNode: child,
+			blockNodes: []
+		};
```

Comment:
> Do we really want developers creating regular expressions and grammars to extract values from a markup string?

WordPress already has the shortcode API, and the parsing of the shortcode could be passed in as state or requested once it is loaded. What other values really need parsing?